internal/pkg/agent/cmd/otel_flags.go

```diff
@@ -27,7 +27,7 @@ func setupOtelFlags(flags *pflag.FlagSet) {
 		" single location can be set per flag entry e.g. `--config=file:/path/to/first --config=file:path/to/second`.")
 
 	flags.StringArray(otelSetFlagName, []string{}, "Set arbitrary component config property. The component has to be defined in the config file and the flag"+
-		" has a higher precedence. Array config properties are overridden and maps are joined. Example --set=processors.batch.timeout=2s")
+		" has a higher precedence. Array config properties are overridden and maps are joined. Example --set \"processors::batch::timeout=2s\"")
```

Comment:
Down below, on line 96, we will actually replace `.` with `::`, which seems wrong given that we cannot use YAML syntax _and_ that prevents using `.` in field names.

Because we are also doing the helper behavior, we _prevent_ allowing `yaml:key: value` syntax because we explicitly require `=` to be present for the `key=value` assignment.